### DISCUSSION TITLE
What is the custom resolver "values" returned key for?

### DISCUSSION THREAD

**Discussion by adriano-tirloni** [react-hook-form/react-hook-form#7275] [Q&A]
Hi! I'm using a custom validation library with custom resolver, I can't figure out the purpose of the `values` object in `{ values, errors }` return;

It doesn't seem to change actual field values after validation or interact with something. Docs only state its default value, what am I missing here?

Thanks!

**Comment by bluebill1049** [comment]
what do you mean by returned key?

**Reply by adriano-tirloni** [reply]
Hi bill! I'm talking about this object:
![Screenshot_20211213-231459_Chrome](https://user-images.githubusercontent.com/6390605/145921388-593da1d1-0e94-4385-90c4-f5dd25a2ad9e.png)
![Screenshot_20211213-231533_Chrome](https://user-images.githubusercontent.com/6390605/145921397-a56f69b7-3871-42a3-a24d-240a8c5ea0c0.png)

**Reply by bluebill1049** [reply]
yes, this `values` will be the returned value for your submission.

**Reply by adriano-tirloni** [reply]
Will it alter the form state? I was playing around with this object, returning some  manually set  values, but it did not alter the fields values after validation ran, or anything else.

I tested it with controlled and uncontrolled fields, nothing changed.

That's why I asked this, I didnt understand ita purpose.

**Reply by bluebill1049** [reply]
provide a simple codesandbox.

**Comment by adriano-tirloni** [comment]
Sure, I will set some time for it.
